From d4720a81ba29902889b2d728b65a00e633067e38 Mon Sep 17 00:00:00 2001
From: Sebastian Lauwers <sebastian.lauwers@interact-iv.com>
Date: Mon, 3 Mar 2014 01:29:27 +0100
Subject: [PATCH 3/3] Added print_function import from __future__

In order to preserve compatibility with Python 2.6 and 2.7, add the
following line as the first line in each file which uses the print()
function.

    from __future__ import print_function

As documented in PEP #236 [1], future_statements always need to be the
first thing that happens in the file.

[1]: http://legacy.python.org/dev/peps/pep-0236/
---
 twisted/test/crash_test_dummy.py |    2 +-
 twisted/test/iosim.py            |    2 ++
 twisted/test/process_cmdline.py  |    2 ++
 twisted/test/process_linger.py   |    1 +
 twisted/test/process_signal.py   |    1 +
 twisted/test/test_banana.py      |    1 +
 twisted/test/test_dirdbm.py      |    1 +
 twisted/test/test_doc.py         |    1 +
 twisted/test/test_internet.py    |    1 +
 twisted/test/test_newcred.py     |    1 +
 twisted/test/test_paths.py       |    1 +
 twisted/test/test_persisted.py   |    1 +
 twisted/test/test_threads.py     |    2 ++
 13 files changed, 16 insertions(+), 1 deletion(-)

diff --git twisted/test/crash_test_dummy.py twisted/test/crash_test_dummy.py
index f99972d..4295d28 100644
--- twisted/test/crash_test_dummy.py
+++ twisted/test/crash_test_dummy.py
@@ -2,7 +2,7 @@
 # Copyright (c) Twisted Matrix Laboratories.
 # See LICENSE for details.
 
-
+from __future__ import print_function
 from twisted.python import components
 from zope.interface import implements, Interface
 
diff --git twisted/test/iosim.py twisted/test/iosim.py
index f077795..a486526 100644
--- twisted/test/iosim.py
+++ twisted/test/iosim.py
@@ -6,6 +6,8 @@
 Utilities and helpers for simulating a network
 """
 
+from __future__ import print_function
+
 import itertools
 
 try:
diff --git twisted/test/process_cmdline.py twisted/test/process_cmdline.py
index b4f80cc..996fd0a 100644
--- twisted/test/process_cmdline.py
+++ twisted/test/process_cmdline.py
@@ -2,6 +2,8 @@
 Write to stdout the command line args it received, one per line.
 """
 
+from __future__ import print_function
 import sys
+
 for x in sys.argv[1:]:
     print(x)
diff --git twisted/test/process_linger.py twisted/test/process_linger.py
index beb9c78..20f0fdc 100644
--- twisted/test/process_linger.py
+++ twisted/test/process_linger.py
@@ -3,6 +3,7 @@ Write to a file descriptor and then close it, waiting a few seconds before
 quitting. This serves to make sure SIGCHLD is actually being noticed.
 """
 
+from __future__ import print_function
 import os, sys, time
 
 print("here is some text")
diff --git twisted/test/process_signal.py twisted/test/process_signal.py
index d3e20cd..99b6660 100644
--- twisted/test/process_signal.py
+++ twisted/test/process_signal.py
@@ -1,3 +1,4 @@
+from __future__ import print_function
 import sys, signal
 
 signal.signal(signal.SIGINT, signal.SIG_DFL)
diff --git twisted/test/test_banana.py twisted/test/test_banana.py
index b5cbbb9..20d2e4b 100644
--- twisted/test/test_banana.py
+++ twisted/test/test_banana.py
@@ -1,6 +1,7 @@
 # Copyright (c) Twisted Matrix Laboratories.
 # See LICENSE for details.
 
+from __future__ import print_function
 import StringIO
 import sys
 
diff --git twisted/test/test_dirdbm.py twisted/test/test_dirdbm.py
index d017831..4f1aae3 100644
--- twisted/test/test_dirdbm.py
+++ twisted/test/test_dirdbm.py
@@ -5,6 +5,7 @@
 Test cases for dirdbm module.
 """
 
+from __future__ import print_function
 import os, shutil, glob
 
 from twisted.trial import unittest
diff --git twisted/test/test_doc.py twisted/test/test_doc.py
index f00f492..fdfab33 100644
--- twisted/test/test_doc.py
+++ twisted/test/test_doc.py
@@ -1,6 +1,7 @@
 # Copyright (c) Twisted Matrix Laboratories.
 # See LICENSE for details.
 
+from __future__ import print_function
 import inspect, glob
 from os import path
 
diff --git twisted/test/test_internet.py twisted/test/test_internet.py
index 1918a87..c697bfe 100644
--- twisted/test/test_internet.py
+++ twisted/test/test_internet.py
@@ -5,6 +5,7 @@
 Tests for lots of functionality provided by L{twisted.internet}.
 """
 
+from __future__ import print_function
 import os
 import sys
 import time
diff --git twisted/test/test_newcred.py twisted/test/test_newcred.py
index b03cd70..a71bae7 100644
--- twisted/test/test_newcred.py
+++ twisted/test/test_newcred.py
@@ -5,6 +5,7 @@
 Tests for L{twisted.cred}, now with 30% more starch.
 """
 
+from __future__ import print_function
 
 import hmac
 from zope.interface import implements, Interface
diff --git twisted/test/test_paths.py twisted/test/test_paths.py
index fc15eee..46c4b63 100644
--- twisted/test/test_paths.py
+++ twisted/test/test_paths.py
@@ -5,6 +5,7 @@
 Test cases covering L{twisted.python.filepath} and L{twisted.python.zippath}.
 """
 
+from __future__ import print_function
 import os, time, pickle, errno, zipfile, stat
 
 from twisted.python.compat import set
diff --git twisted/test/test_persisted.py twisted/test/test_persisted.py
index e0f0868..7197dda 100644
--- twisted/test/test_persisted.py
+++ twisted/test/test_persisted.py
@@ -2,6 +2,7 @@
 # Copyright (c) Twisted Matrix Laboratories.
 # See LICENSE for details.
 
+from __future__ import print_function
 
 # System Imports
 import sys
diff --git twisted/test/test_threads.py twisted/test/test_threads.py
index 9148e9a..324661a 100644
--- twisted/test/test_threads.py
+++ twisted/test/test_threads.py
@@ -321,6 +321,8 @@ class DeferToThreadPoolTestCase(unittest.TestCase):
 
 
 _callBeforeStartupProgram = """
+from __future__ import print_function
+
 import time
 import %(reactor)s
 %(reactor)s.install()
-- 
1.7.10.4

